projects/demo/src/app/pages/examples/various/virtual-scroll/advanced-example-virtual-scroll.component.ts

```diff
@@ -0,0 +1,282 @@
+      "url": "https://via.placeholder.com/600/b0f7cc",
+      "thumbnailUrl": "https://via.placeholder.com/150/b0f7cc"
+    },
+    {
+      "albumId": 1,
+      "id": 8,
+      "title": "aut porro officiis laborum odit ea laudantium corporis",
+      "url": "https://via.placeholder.com/600/54176f",
+      "thumbnailUrl": "https://via.placeholder.com/150/54176f"
+    },
+    {
+      "albumId": 1,
+      "id": 9,
+      "title": "qui eius qui autem sed",
+      "url": "https://via.placeholder.com/600/51aa97",
+      "thumbnailUrl": "https://via.placeholder.com/150/51aa97"
+    },
+    {
+      "albumId": 1,
+      "id": 10,
+      "title": "beatae et provident et ut vel",
+      "url": "https://via.placeholder.com/600/810b14",
+      "thumbnailUrl": "https://via.placeholder.com/150/810b14"
+    },
+  ];
+  public readonly source: ServerDataSource;
+  public readonly settingsServer: Settings = {
+    virtualScroll: {
+      itemSize: 40,
+      minBufferPx: 100,
+      maxBufferPx: 100,
+      viewportHeight: '300px',
+      infiniteScroll: {
+        getNextFunction: (offset: number) => this.getNextBatchServer(offset),
+        threshold: 2,
+      }
+    },
+    pager: {
+      display: false,
+      perPage: Number.MAX_SAFE_INTEGER,
+    },
+    columns: {
+      id: {
+        title: 'ID',
+      },
+      albumId: {
+        title: 'Album',
+      },
+      title: {
+        title: 'Title',
+      },
+      url: {
+        title: 'Url',
+      },
+    },
+  };
+
+  constructor(http: HttpClient) {
+    this.data = this.generateFakeData();
+    this.source = new ServerDataSource(http, {endPoint: 'https://jsonplaceholder.typicode.com/photos'});
+  }
+
+  private generateFakeData(): any[] {
+    const data = [];
+    for (let i = 0; i < this.fakeNextLimit; i++) {
+      data.push({
+        id: i + 1,
+        name: this.fakeDataSetLocal[i % this.fakeDataSetLocal.length].name,
+        username: this.fakeDataSetLocal[i % this.fakeDataSetLocal.length].username,
+        email: this.fakeDataSetLocal[i % this.fakeDataSetLocal.length].email,
+      });
+    }
+    return data;
+  }
+
+  private getNextBatch(offset: number,): Observable<any[]> {
+    const nextData = [];
+    const nextLimit = Math.min(this.fakeNextLimit, this.fakeNextDataLength - offset);
+    this.loadingPlaceholdersLocal = new Array(nextLimit);
+
+    for (let i = 0; i < nextLimit; i++) {
+      nextData.push({
+        id: (offset + (i + 1)).toString(),
+        name: this.fakeDataSetLocal[i % this.fakeDataSetLocal.length].name,
+        username: this.fakeDataSetLocal[i % this.fakeDataSetLocal.length].username,
+        email: this.fakeDataSetLocal[i % this.fakeDataSetLocal.length].email,
+      });
+    }
+
+    return of(nextData).pipe(
+      delay(this.fakeDelay),
+      tap(() => this.loadingPlaceholdersLocal = []),
+    );
+  }
+
+  private getNextBatchServer(offset: number): Observable<any[]> {
```

Comment:
I do not understand this function. What is it supposed to do? It is never called, until you reach the end of the table. Also, the "fake data" is not coming from the server, which looks like quite disconnected from what a user would actually need to do. So I don't think this serves well as an _example_.

The second observation is: when the end of the table (i.e. after 5000 entries) is reached, 20 identical HTTP requests are sent out (https://jsonplaceholder.typicode.com/photos?_sort=&_order=&_page=1&_limit=9007199254740991).

Maybe the infinite scroll is not working as inteded. Please check.